# <font color='gold'>**_Análise Exploratória Descritiva: Metacritc Score - Video Games_**</font> 🎮


<kbd><img src="images\gamer_stock.jpg" width=1000 height=350></kbd>
---

# <font color='gold'>**_Objetivos_**</font>

---

Nesse projeto, iremos realizar uma análise exploratória descritiva de dados em um [Dataset](https://www.kaggle.com/datasets/patkle/metacritic-scores-for-games-movies-tv-and-music) que contém avaliações fornecidas pelo [Metacritic](https://www.metacritic.com/game) sobre diversos títulos de Video-Game. Como um sumário do que será realizado nesse projeto, temos:

<font color='gold'>**_1._**</font> **Compreender os Dados**:
- Aqui será realizada uma análise estatística-descritiva dos dados, buscando compreender seu formato e suas características centrais.

- Também iremos utilizar de Histogramas e Boxplots/Violinplots para visualizarmos a distribuição dos dados, outliers e etc.

<font color='gold'>**_2._**</font> **Limpeza, Correção e Tratamento**:
- Realizaremos a limpeza e tratamento desses dados. Com as informações adiquiridas na etapa anterior, iremos buscar adaptar e refatorar o Dataset para o melhor formato possível.

- Utilizaremos funções como principal maneira de limpeza e tratamento.

<font color='gold'>**_3._**</font> **Análise Exploratória e Visualizações**:
- Após a compreensão e tratamento desses dados, iremos realizar uma análise exploratória, buscando gerar insights e a retirada de informações relevantes e curiosas dos dados.

- Para isso, utilizaremos de diversas visualizações utilizando Matplotlib e Seaborn.

# <font color='gold'>**_Importando Pacotes e Compreendendo Dados_**</font>

---

In [11]:
# Importando bibliotecas necessárias para análise:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
from platform import python_version

# Setando estilo escuto do grid para gráficos Seaborn:
sns.set_theme(context='notebook', style='darkgrid')

# Filtrando avisos de erro:
warnings.filterwarnings('ignore')

# Mostrando versões utilizadas:
print(' Pacotes Prontos! Versões Utilizadas: '.center(70, '-'))
print(f'Python:     --> {python_version()}')
print(f'Pandas:     --> {pd.__version__}')
print(f'Numpy:      --> {np.__version__}')
print(f'Matplotlib: --> {matplotlib.__version__}')
print(f'Seaborn:    --> {sns.__version__}')


---------------- Pacotes Prontos! Versões Utilizadas: ----------------
Python:     --> 3.9.16
Pandas:     --> 1.5.2
Numpy:      --> 1.23.5
Matplotlib: --> 3.6.2
Seaborn:    --> 0.11.2


In [13]:
# Criando o Dataframe:
data_raw = pd.read_csv('data/games.csv', sep=',')

In [14]:
# Visualizando primeiros registros:
data_raw.head(10)

id  metascore       platform       release_date  sort_no  \
0  543718         91             PC    August 18, 2020      301   
1  555108         91             PC   December 8, 2022      302   
2  106820         91  PlayStation 2   November 7, 2005      303   
3  110775         91             PC  November 13, 2008      304   
4  142864         91            Wii   October 26, 2010      305   
5  114951         91            Wii    August 24, 2009      306   
6  397035         91             PC       May 23, 2016      307   
7  102812         91             PC      June 24, 2003      308   
8  161540         91       GameCube   October 23, 2002      309   
9  100705         91  PlayStation 2   October 23, 2000      310   

                                             summary  \
0  From light planes to wide-body jets, fly highl...   
1  Take up your sword, channel your magic or boar...   
2  Strap on your Guitar Hero SG controller, plug-...   
3  Players last visited Northrend in "Warcraft II...   
4  Rock Band returns with the third iteration of ...   
5  Metroid Prime 3: Corruption set a new standard...   
6  Overwatch is a highly stylized team-based shoo...   
7  Cutting-edge technology brings you the most ac...   
8  Drop into the ultimate challenge.  In the next...   
9  EA Sports steps up to the challenge of Sega's ...   

                                       title user_score  
0                 Microsoft Flight Simulator        7.1  
1                             Chained Echoes        8.7  
2                                Guitar Hero        8.5  
3  World of Warcraft: Wrath of the Lich King        7.7  
4                                Rock Band 3        6.8  
5                      Metroid Prime Trilogy          9  
6                                  Overwatch        6.4  
7                       F1 Challenge '99-'02        7.7  
8                   Tony Hawk's Pro Skater 4        8.2  
9                            Madden NFL 2001        7.5

In [15]:
# Visualizando um sample aleatório dos dados:
data_raw.sample(10)

id  metascore          platform        release_date  sort_no  \
10690  500609         72               3DS     October 6, 2017    10691   
17791  102204         55                PC     August 13, 2002    17792   
792    101304         88                PC     August 27, 2002      793   
4893   346665         80                PC    November 7, 2014     4894   
13434  564964         67                PC   December 16, 2021    13435   
58     100009         91                PC     October 9, 1997      359   
3821   459728         81               3DS      April 16, 2015     3822   
7950   521309         75     PlayStation 4  September 24, 2019     7951   
7512   214036         76          Xbox 360       July 11, 2012     7513   
12674  497970         68  PlayStation Vita    October 18, 2016    12675   

                                                 summary  \
10690  A new adventure in the Layton series –with a f...   
17791  On a routine hunt on Planet LV1201 a lone Pred...   
792    Mafia brings the 1930's underworld to life in ...   
4893   Football Manager 2015 allows you to live the l...   
13434  The Gunk is a game about two friends who run a...   
58     In Dark Forces, Kyle Katarn, a young mercenary...   
3821   3D Fantasy Zone II W is the side-scrolling sho...   
7950   [OVERCOME]\nOn the way to Jericho City, your p...   
7512   In Zuma the player takes the role of a fearles...   
12674  An RPG in which Kanata and others have been im...   

                                                   title user_score  
10690  Layton's Mystery Journey: Katrielle and The Mi...        6.5  
17791  Aliens Versus Predator 2: Primal Hunt Expansio...        7.3  
792                                                Mafia        9.1  
4893                               Football Manager 2015        6.2  
13434                                           The Gunk        4.8  
58                 Star Wars Jedi Knight: Dark Forces II        8.6  
3821                                3D Fantasy Zone II W          7  
7950                                         The Surge 2        7.1  
7512                                     Zuma's Revenge!        7.1  
12674           Exist Archive: The Other Side of the Sky        8.1

Com a visualização geral dos dados podemos nos aprofundar mais em suas características:

In [19]:
# Checando dimensões dos dados:
data_raw.shape

(20022, 8)

Temos um dataset relativamente grande, contendo <font color='gold'>**_20.022_ registros**</font> em <font color='gold'>**_8 variáveis_**</font> diferentes.

Vamos checar informações mais específicas acerca dessas variáveis:

In [20]:
# Visão mais detalhada das variáveis:
data_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20022 entries, 0 to 20021
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            20022 non-null  int64 
 1   metascore     20022 non-null  int64 
 2   platform      20022 non-null  object
 3   release_date  20022 non-null  object
 4   sort_no       20022 non-null  int64 
 5   summary       19930 non-null  object
 6   title         20022 non-null  object
 7   user_score    20022 non-null  object
dtypes: int64(3), object(5)
memory usage: 1.2+ MB


Informações relevantes:

- Temos alguns valores faltantes apenas na coluna **`summary`**, o que indica que o Dataset está bem completo. Também, esse fator não é um problema por a coluna **`summary`** não será utilizada nas análises.

- Sobre os Data types utilizados: 

    - iremos avaliar mudanças possíveis, como int32 em vez de int64 nas colunas que possuem esse data type. 

    - A coluna **`release_date`** será trasnformada em uma coluna do tipo date em vez de object.

    - A coluna **`user_score`** está no formato object quando deveria estar em float, por se tratar da variável que contém as avaliações fornecidas pelos usuários.
    
Com essas mudanças, a memória consumida será reduzida. Nesse dataset não fará uma diferença notável na performance, porém é uma boa prática sempre deixar os dados no formato mais eficiente possível.

## **Análise Estatística - Descritiva dos Dados**

Conseguimos coletar informações relevantes sobre os dados na etapa anterior, porém para podermos ter uma visão realmente aprofundada desses dados, vamos realizar uma análise estatística para descrever esse conjunto.

- Resumo Estatístico das Variáveis Numéricas:

In [23]:
# Utilizando método describe nas colunas numéricas:
data_raw.describe()

id     metascore       sort_no
count   20022.000000  20022.000000  20022.000000
mean   296058.651084     70.758566  10011.500000
std    190522.456059     12.165042   5779.997881
min    100002.000000     11.000000      1.000000
25%    112180.250000     64.000000   5006.250000
50%    188870.500000     72.000000  10011.500000
75%    510982.250000     80.000000  15016.750000
max    626054.000000     99.000000  20022.000000

Nesse dataset temos três variáveis numéricas, porém apenas **`metascore`** nos importa para essa análise, pois as outras duas variáveis são **`id`** e **`sort_no`** não contém informações relevantes por se tratarem de variáveis de organização dos dados (id, clasificação, etc).

Sobre a coluna **`metascore`**:

- A média geral da avaliação do Metacritic para os jogos (lembrando que essa avaliação vai de 0 a 100) é de <font color='gold'>**70.8**</font>, uma média bem alta considerando o limite de 100.

- O desvio padrão dessas avaliações é de  <font color='gold'>**12.2**</font> pontos acima ou abaixo da média.

- A avaliação mais baixa presente nos dados é de  <font color='gold'>**11**</font>, realmente um valor bem baixo.

- Já a avaliação mais alta é de  <font color='gold'>**99**</font>, aparentemente nenhum game recebeu a nota máxima de 100.

Sobre os quartis e percentis:

- 25% das avaliações do Metacritic estão abaixo ou são iguais a  <font color='gold'>**64**</font>;

- 50% das avaliações do Metacritic estão abaixo ou são iguais a  <font color='gold'>**72**</font>, esse valor representa a mediana das avaliações, ou seja, o valor central dos dados ordenados é 72;

- 75% das avaliações do metacritic estão abaixo ou são iguais a  <font color='gold'>**80**</font>;

- resumo Estatístico das Variáveis Categóricas:

In [27]:
# Utilizando método describe nas colunas categóricas:
data_raw.describe(exclude='number')

platform       release_date  \
count     20022              20022   
unique       22               4794   
top          PC  November 14, 2006   
freq       5276                 48   

                                                  summary          title  \
count                                               19930          20022   
unique                                              15545          12926   
top     Iron Man, an explosive third-person action sho...  Madden NFL 07   
freq                                                    7              9   

       user_score  
count       20022  
unique         94  
top           tbd  
freq         1382

Das variáveis categóricas, **`release_date`** e **`user_score`** não são apropriadas para essa análise, pois deveriam estar em formato date e float respectivamente. Sendo assim, vamos olhar para as variáveis realmente categóricas nos dados:

**`platform`**

- Temos <font color='gold'>**22 plataformas diferentes**</font> nos dados. Isso indica uma variedade grande nas tecnologias utilziadas para games como consoles ao longo do tempo.

- A plataforma em que mais jogos estão presentes é o  <font color='gold'>**PC**</font>, contando com  <font color='gold'>**5.276 registros**</font> nos dados.

**`title`**

- Temos <font color='gold'>**12.926 títulos únicos nos dados**</font>. Isso indica que dos 20.022 títulos totais, 7.096 deles podem ser o mesmo jogo porém em um registro diferente para cada plataforma em que esse jogo foi lançado. Isso será explorado mais a frente.

Como dito previamente, a coluna `summary` não será utilizada, então sua descrição estatística não nos importa.